src/lib.rs

```diff
@@ -19,28 +20,89 @@ pub fn derive_key(
     Box::new(keypair.to_bytes())
 }
 
-/// derive_key_c does the same thing as the above function, but is intended for use over the CFFI.
-/// note that the caller must free() the returned memory as it's not managed/freed here.
+// check for error. if no error, do nothing. if there is an error, print it and return a null ptr.
```

Comment:
are you sure that this is good option to merge? in smcli this is probably will be seen, but inside smapp it won't be even written as a log. and beside that, will likely fail with some undefined behavior

Reply:
ah sorry you kept original function as is